src/Stratis.Bitcoin/P2P/PeerDiscoveryLoop.cs

```diff
@@ -106,47 +114,57 @@ public void DiscoverPeers(IConnectionManager connectionManager)
         }
 
         /// <summary>
-        /// See <see cref="DiscoverPeers"/>.
+        /// See <see cref="DiscoverPeers"/>. This loop deals with discovery from DNS seeds and seed nodes as opposed to peers.
         /// </summary>
-        private async Task DiscoverPeersAsync()
+        private async Task DiscoverFromDnsSeedsAsync()
         {
             var peersToDiscover = new List<IPEndPoint>();
-            List<PeerAddress> foundPeers = this.peerAddressManager.PeerSelector.SelectPeersForDiscovery(1000).ToList();
-            peersToDiscover.AddRange(foundPeers.Select(p => p.Endpoint));
+
+            // First see if we need to do DNS discovery at all. We may have peers from a previous cycle that still need to be tried.
+            if (this.peerAddressManager.Peers.Select(a => !a.Attempted).Any())
```

Comment:
So if there are any peers that have been attempted we do discovery from dns/seeds?

I think there is another `if` missing here?

Reply:
No, if there are no peers that have not been attempted then we do discovery from DNS/seeds.

Reply:
Note that 'Attempted' means 'tried and failed'.

Reply:
ok this I agree, basically this means we have exhausted all attempts